flink-metrics/flink-metrics-core/src/main/java/org/apache/flink/metrics/MetricGroup.java

```diff
@@ -189,4 +189,15 @@
 	 * @return fully qualified metric name
 	 */
 	String getMetricIdentifier(String metricName, CharacterFilter filter);
+
+	/**
+	 * Returns the fully qualified metric name for reporter with certain number position in group, for example
+	 * {@code "host-7.taskmanager-2.window_word_count.my-mapper.metricName"}
+	 *
+	 * @param metricName metric name
+	 * @param filter character filter which is applied to the scope components if not null.
+	 * @param indexReporter position number of reporter in group reporters
+	 * @return fully qualified metric name
+	 */
+	String getMetricIdentifier(String metricName, CharacterFilter filter,int indexReporter);
```

Comment:
This method should be part of `AbstractMetricGroup`. The signature of the `MetricRegistry#register()` may be changed if necessary.